### PR TITLE
Export executor project

### PR DESCRIPTION
**Pull request by qris**
See [JENKINS-67651](https://issues.jenkins.io/browse/JENKINS-67651).

### Proposed changelog entries

* Entry 1: Expose project name building on each executor via computer objects in the HTTP APIs
* ...

### Proposed upgrade guidelines

N/A

### Submitter checklist

- [X] (If applicable) Jira issue is well described
- [X] Changelog entries and upgrade guidelines are appropriate for the audience affected by the change (users or developer, depending on the change). [Examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)
  * Fill-in the `Proposed changelog entries` section only if there are breaking changes or other changes which may require extra steps from users during the upgrade
- [ ] Appropriate autotests or explanation to why this change has no tests
- [ ] For dependency updates: links to external changelogs and, if possible, full diffs

### Desired reviewers

@mention

### Maintainer checklist

Before the changes are marked as `ready-for-merge`: 

- [ ] There are at least 2 approvals for the pull request and no outstanding requests for change
- [ ] Conversations in the pull request are over OR it is explicit that a reviewer does not block the change
- [ ] Changelog entries in the PR title and/or `Proposed changelog entries` are correct
- [ ] Proper changelog labels are set so that the changelog can be generated automatically
- [ ] If the change needs additional upgrade steps from users, `upgrade-guide-needed` label is set and there is a `Proposed upgrade guidelines` section in the PR title. ([example](https://github.com/jenkinsci/jenkins/pull/4387))
- [ ] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins-ci.org/issues/?filter=12146)).

### CODE REVIEW

**Review by daniel-beck — COMMENTED** [review]
I don't think this change in isolation makes sense, as it only works for `AbstractBuild` and ignores non-`AbstractProject` job types.

Additionally, I'm not convinced the API doesn't already provide this information based in the description in Jira. I recommend you go exploring with `?depth=`.

File: core/src/main/java/hudson/model/AbstractBuild.java [hunk 1]
```diff
@@ -895,7 +896,7 @@ public boolean tearDown(AbstractBuild build, BuildListener listener) throws IOEx
      *
      * @return never null.
      */
-    @Exported
+        @Exported
```
Comment (daniel-beck) [inline core/src/main/java/hudson/model/AbstractBuild.java:899]: Unintentional whitespace change, revert.

File: core/src/main/java/hudson/model/Project.java [hunk 2]
```diff
@@ -107,6 +108,11 @@ public void onLoad(ItemGroup<? extends Item> parent, String name) throws IOExcep
     @Override public Item asItem() {
         return this;
     }
+    
+    @Exported
+    public String getName() {
+        return Item.getName();
```
Comment (daniel-beck) [inline core/src/main/java/hudson/model/Project.java:114]: Not a useful response as soon as folders are involved.

**Comment by qris** [comment]
Thanks for the quick response! Using the depth parameter I found out about the `currentExecutable` object which seems to help somewhat. I can get the `fullDisplayName` which contains the job name (I guess, perhaps its display name which might be different?) and build number, but combined, so I have to parse it. Not difficult, but it would be cleaner to be able to retrieve these separately. Perhaps I can just make the components that go into that @Exported, would that be acceptable?

File: core/src/main/java/hudson/model/Project.java (hunk 2 above)
Comment (mawinter69) [reply core/src/main/java/hudson/model/Project.java:114]: Project is inheriting from AbstractItem that already exports getName which is the implementation of Item.getName. So this is completely nonsense. That code isn't compiling

File: core/src/main/java/hudson/model/AbstractBuild.java [hunk 3]
```diff
@@ -176,6 +176,7 @@ protected AbstractBuild(P project, File buildDir) throws IOException {
         super(project, buildDir);
     }
 
+    @Exported
```
Comment (mawinter69) [inline core/src/main/java/hudson/model/AbstractBuild.java:179]: This limits it to things that inherit from AbstractBuild. Meaning all pipeline jobs will not appear.
Maybe put it on the getParent of the Run class

**Review by mawinter69 — CHANGES_REQUESTED** [review]

**Comment by alecharp** [comment]
@qris where you able to see the latest comment from Daniel on this pull request? Are you still interested to get its content included in Jenkins core?

**Comment by alecharp** [comment]
Thanks! Great that you could have a solution.

**Pull request closed without merging**